app/src/main/java/com/github/sdpcoachme/auth/LoginActivity.kt

```diff
@@ -94,20 +118,49 @@ class LoginActivity : ComponentActivity() {
         }
     }
 
+    // Declare the launcher at the top of your Activity/Fragment:
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted: Boolean ->
+        if (isGranted) {
+            Toast.makeText(this, "Notifications enabled", Toast.LENGTH_SHORT).show()
+        } else {
+            Toast.makeText(this, "Notifications disabled", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    /**
+     * Asks the user for permission to post notifications.
+     */
+    private fun askNotificationPermission() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+            // As discussed on discord with Léo and Kamilla,
+            // for API level < 33, notifications cannot be enabled / disabled in the app
+            // and the user needs to disable them in the settings.
+            return
+        }
+
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS)
+            == PackageManager.PERMISSION_GRANTED) {
+            // Notifications have already been enabled, no need to ask again.
+            return
+        }
+
+        // Request permission to post notifications.
+        requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+    }
+
     private fun launchPostLoginActivity(email: String) {
         database.userExists(email).thenAccept { exists ->
             if (exists) {
-                launchActivity(MapActivity::class.java)
+                startActivity(userExistsIntent)
             } else {
-                launchActivity(SignupActivity::class.java)
+                startActivity(Intent(this, SignupActivity::class.java))
             }
-        }
 
-    }
-
-    private fun launchActivity(activity: Class<*>) {
-        val intent = Intent(this, activity)
-        startActivity(intent)
+            askNotificationPermission()
```

Comment:
Here, I thought that it might make more sense to keep the method call in the LoginActivity in case a future iteration of the program does not always open the MapActivity after logging in. However, I am not sure what makes more sense.

Reply:
Yes, we can keep it like this for now!